services/brig/src/Brig/API/User.hs

```diff
@@ -346,23 +348,37 @@ createUser new@NewUser {..} = do
       Team.TeamName nm <- lift $ Intra.getTeamName tid
       pure $ CreateUserTeam tid nm
 
+makeUserPermanent :: (Team.Invitation, invInfo) -> AppIO ()
+makeUserPermanent (coerce . Team.inInvitation . fst -> uid) = do
+  mbAccount <- Data.lookupAccount uid
+  for_ mbAccount $ \account -> do
+    Data.insertAccount account Nothing Nothing True
+    makeHandlePermanent `mapM_` userHandle (accountUser account)
+
+  mbRichInfo <- Data.lookupRichInfo uid
+  for_ mbRichInfo $ Data.updateRichInfo uid
+
 -- | 'createUser' is becoming hard to maintian, and instead of adding more case distinctions
 -- all over the place there, we add a new function that handles just the one new flow where
 -- users are invited to the team via scim.
 createUserInviteViaScim :: UserId -> NewUserScimInvitation -> ExceptT Error.Error AppIO UserAccount
-createUserInviteViaScim uid (NewUserScimInvitation tid loc name rawEmail) = (`catchE` (throwE . Error.newUserError)) $ do
-  email <- either (throwE . InvalidEmail rawEmail) pure (validateEmail rawEmail)
+createUserInviteViaScim uid (NewUserScimInvitation tid loc name rawHandl rawEmail richinfo) = do
+  email <- either (throwE . Error.newUserError . InvalidEmail rawEmail) pure (validateEmail rawEmail)
+  handl <- validateHandle (fromHandle rawHandl)
   let emKey = userEmailKey email
-  verifyUniquenessAndCheckBlacklist emKey
-  account <- lift $ newAccountInviteViaScim uid tid loc name email
+  verifyUniquenessAndCheckBlacklist emKey `catchE` (throwE . Error.newUserError)
+  account <- lift $ Data.newAccountInviteViaScim uid tid loc name email
   Log.debug $ field "user" (toByteString . userId . accountUser $ account) . field "action" (Log.val "User.createUserInviteViaScim")
   let activated =
         -- It would be nice to set this to 'False' to make sure we're not accidentally
         -- treating 'PendingActivation' as 'Active', but then 'Brig.Data.User.toIdentity'
         -- would not produce an identity, and so we won't have the email address to construct
         -- the SCIM user.
         True
```

Comment:
As we discussed a few weeks ago, setting this to true is problematic for other kinds of queries. Can you not simply write a slightly different accountLookup query to just extract the pendingInvitation users which doesn't filter on activated?

Reply:
That would have been an option, but I'm not sure it would result in better code structure.  As it is now, we went through all calls to the existing lookup functions transitively, and found several places where we want to handle pending invitations as well (eg. update, delete).

I'm pretty confident about the current solution.  How badly to you want to fight this battle?  Would it make sense to have a call and discuss it for half an hour or so, looking at the code?

Reply:
There are some things possible now that shouldn't be possible: I can search for, and create a connection request with a person who is "pendingInvitation". But I'm not sure the person joining can then see this connection request after joining. I can also easily create hundreds of accounts that are active without ever validating any emails, so this is prone to spam. There might be some other problematic cases here.

Reply:
> I can search for, and create a connection request with a person who is "pendingInvitation".

What's your evidence?  I think we wrote a test case that makes sure this is not the case.

> Note that the current PR attempts to use a delete-account-via-TTL. But an account creates more things than just one row in the users table.

Did you read this PR?  (Not just the first few commits, I mean.  The whole thing.)

Reply:
I admit I didn't look all your code changes; perhaps I don't need to be worried and all the edge cases are covered. Sorry, it's not helpful for me to make comments without having looked in detail. So I'll either stop reviewing this PR, or review it in-depth before adding more comments. With my attention elsewhere at the moment, I think I'll trust you to do the right thing :)